judge/build_executable.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+
+# Script to build executables.
+#
+# Usage: $0 <workdir>
+
+set -e
+trap error EXIT
+
+cleanexit ()
+{
+	trap - EXIT
+
+	$DJ_LIBJUDGEDIR/chroot-startstop.sh stop
+
+	# Make sure that all files are owned by the current user/group, so
+	# that we can delete the judging output tree without root access.
+	# We also remove group RUNGROUP so that this can safely be shared
+	# across multiple judgedaemons, and remove write permissions.
+	$GAINROOT chown -R "$(id -un):" $WORKDIR
+	chmod -R go-w $WORKDIR
+	for i in bin etc lib lib64 proc usr; do
+		rm -rf $i
```

Comment:
Why is this necessary? It happened to me that `proc` was not unmounted (probably because of an error in a previous run), and then I got an endless list of permission denied errors (and maybe some files deleted).

Reply:
Because otherwise the (empty) dirs are also copied into each judging dir which clutters things.

Reply:
Can't we build/compile in a subdirectory and only copy contents of that directory? (I haven't looked into all the details yet of your PR.)

Reply:
This is indeed possible I think